src/string_lib.c

```diff
@@ -1,5 +1,7 @@
 #include <stdio.h>
 
+int EOS = '\0';
```

Comment:
As it is, you just declared a global variable!
You should either add the `static` & `const` keywords or transform this into a macro.

```suggestion
static const int EOS = '\0';
```
or
```suggestion
#define EOS '\0'
```

`static` tells the compiler that this variable is only to be accessed by this compilation unit (file).
`const` has pretty much the same meaning as in JS.

WARNING: `EOS` might mean `End Of Stream` in other contexts.

Reply:
Thanks a lot man ^^